homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,250 @@
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data.get(CONF_USERNAME),
+                data.get(CONF_PASSWORD),
+                data.get(MQTT_TRANSPORT),
+            )
+
+            if test_connect:
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: data[CONF_HOST],
+                        CONF_PORT: data[CONF_PORT],
+                        CONF_USERNAME: data.get(CONF_USERNAME),
+                        CONF_PASSWORD: data.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: data.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
+                    },
+                )
+
+            errors["base"] = "cannot_connect"
+
+        return self.async_show_form(
+            step_id="confirm",
+            description_placeholders={"addon": self._hassio_discovery["addon"]},
+            errors=errors,
+        )
+
+
+class InelsOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle iNELS options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.broker_config: dict[str, str | int] = {}
+        self.options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input: None = None) -> FlowResult:
+        """Manage the MQTT setup."""
+        return await self.async_step_setup()
+
+    async def async_step_setup(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the MQTT configuration to connect."""
+        errors = {}
+        current_config = self.config_entry.data
+
+        if user_input is not None:
+            test_connect = await self.hass.async_add_executor_job(
+                try_connection,
+                self.hass,
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                user_input.get(CONF_USERNAME),
+                user_input.get(CONF_PASSWORD),
+                user_input.get(MQTT_TRANSPORT),
+            )
+
+            if test_connect:
+                self.broker_config.update(user_input)
+                self.hass.config_entries.async_update_entry(
+                    self.config_entry, data=self.broker_config
+                )
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_PORT: user_input[CONF_PORT],
+                        CONF_USERNAME: user_input.get(CONF_USERNAME),
+                        CONF_PASSWORD: user_input.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: user_input.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
+                    },
+                )
+
+            errors["base"] = "cannot_connect"
```

Comment:
are you able to differentiate between cannot connect and invalid auth? If so, you should.

As well, you should check for any unknown errors.